compiler/rustc_const_eval/src/interpret/intrinsics.rs

```diff
@@ -418,7 +422,11 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                     )?;
                 }
                 if intrinsic_name == sym::assert_uninit_valid
-                    && !layout.might_permit_raw_init(self, /*zero:*/ false)
+                    && !layout.might_permit_raw_init(
+                        self,
+                        /*zero:*/ false,
+                        self.tcx.sess.opts.debugging_opts.strict_init_checks,
```

Comment:
somewhat annoying to pass this everywhere, when `self` is also passed. Though `HasDataLayout` doesn't have a useful way to get at this.

maybe add a field to [`TargetDataLayout`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_target/abi/struct.TargetDataLayout.html#) for that bool?

Idk, doesn't seem great either. I guess let's roll with the PR as it is.